integtest/src/main/java/org/corfudb/universe/cluster/docker/DockerCluster.java

```diff
@@ -0,0 +1,200 @@
+        });
+
+        // Kill all docker containers
+        clusterParams.getServices().keySet().forEach(serviceName -> {
+            ServiceParams<ServerParams> serviceParams = clusterParams.getService(serviceName);
+
+            serviceParams.getNodes().forEach(serverParams -> {
+                try {
+                    docker.killContainer(serverParams.getGenericName());
+                } catch (DockerException | InterruptedException e) {
+                    log.debug("Can't kill container. In general it should be killed already. Container: {}",
+                            serverParams.getGenericName());
+                }
+            });
+
+        });
+
+        try {
+            network.shutdown();
+        } catch (ClusterException e) {
+            log.debug("Can't stop docker network. Already stopped");
+        }
+    }
+
+    @Override
+    public ImmutableMap<String, Service> services() {
+        return services;
+    }
+
+    @Override
+    public Service getService(String serviceName) {
+        return services.get(serviceName);
+    }
+
+    private Map<String, Service> createAndDeployServices() {
+        Map<String, Service> servicesSnapshot = new HashMap<>();
+
+        for (String serviceName : clusterParams.getServices().keySet()) {
+            DockerService service = DockerService.builder()
+                    .clusterParams(clusterParams)
+                    .params(clusterParams.getService(serviceName))
+                    .docker(docker)
+                    .build();
+
+            service.getParams().getNodes().forEach(node ->
+                    FAKE_DNS.addForwardResolution(node.getGenericName(), InetAddress.getLoopbackAddress())
+            );
+
+            service = service.deploy();
+
+            servicesSnapshot.put(serviceName, service);
+        }
+
+        return servicesSnapshot;
+    }
+
+    private class DockerNetwork {
+        private final Logger log = LoggerFactory.getLogger(DockerNetwork.class);
+
+        /**
+         * Sets up a docker network.
+         *
+         * @throws ClusterException will be thrown if cannot set up a docker network
+         */
+        void setup() throws ClusterException {
+            log.info("Setup network: {}", clusterParams.getNetworkName());
+            NetworkConfig networkConfig = NetworkConfig.builder()
+                    .checkDuplicate(true)
+                    .attachable(true)
+                    .name(clusterParams.getNetworkName())
+                    .build();
+
+            try {
+                docker.createNetwork(networkConfig);
+            } catch (DockerException | InterruptedException e) {
+                throw new ClusterException("Cannot setup docker network.", e);
+            }
+        }
+
+        /**
+         * Shuts down a docker network.
+         *
+         * @throws ClusterException will be thrown if cannot shut up a docker network
+         */
+        void shutdown() throws ClusterException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.corfudb.universe.cluster.ClusterException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)